### PR TITLE
[Stripe] Improved Stripe.tokens.IToken

### PR DESCRIPTION
**Pull request by htunnicliff**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://stripe.com/docs/api/tokens/object
----
I refactored the `IToken` interface to a Union type to better reflect the nature of tokens, in that they are either a `card` or `bank_account`. 

### Existing
```ts
const token = stripe.tokens.retrieve("tok_xxxxx");
if (token.type === "card") {
  console.log(token.card.fingerprint)
  // Error: Object is possibly 'undefined'. ts(2532)
}
```

### With Union Type
```ts
const token = stripe.tokens.retrieve("tok_xxxxx");
if (token.type === "card") {
  console.log(token.card.fingerprint)
  // No error!
}
```

### CODE REVIEW

File: types/stripe/index.d.ts [hunk 1]
```diff
@@ -6977,20 +6977,22 @@ declare namespace Stripe {
     }
 
     namespace tokens {
-        interface IToken extends ICardToken, IBankAccountToken {}
+        type IToken = ICardToken | IBankAccountToken
 
         interface ICardToken extends ITokenBase {
+            type: "card";
```
Comment (pushplay) [inline types/stripe/index.d.ts]: This should be single quotes.  Also there should be an empty line before the next comment.

Same goes for IBankAccountToken.

**Review by pushplay — CHANGES_REQUESTED** [review]
Yup union type is totally the right thing to do here.  Good catch.  Just needs some formatting improvements for consistency and it's good to go.

**Comment by htunnicliff** [comment]
I made formatting changes requested by @pushplay and added some tests as well.

File: types/stripe/index.d.ts [hunk 2]
```diff
@@ -6977,20 +6977,22 @@ declare namespace Stripe {
     }
 
     namespace tokens {
-        interface IToken extends ICardToken, IBankAccountToken {}
+        type IToken = ICardToken | IBankAccountToken;
 
         interface ICardToken extends ITokenBase {
+            type: 'card';
```
Comment (pushplay) [inline types/stripe/index.d.ts]: Still needs a newline here.

**Review by pushplay — CHANGES_REQUESTED** [review]
Thanks for adding the tests!  The newlines still need to be addressed and then it's good to go.

**Comment by htunnicliff** [comment]
Newlines added!

**Pull request closed without merging**